### PR TITLE
Adding is-featured in events model and schema

### PR DESCRIPTION
**Pull request by uds5501**
Fixes fossasia/open-event-frontend#1925 and #5552

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Creates a new field in events model so that featured status of the event can be stored.

#### Changes proposed in this pull request:

- Changed `event` model accordingly
- Using `is_featured` as Boolean field with default value of `False`

### CODE REVIEW

**Comment by uds5501** [comment]
@CosmicCoder96 Please review

**Comment by uds5501** [comment]
@CosmicCoder96 My bad, i shall add that soon

**Comment by uds5501** [comment]
@CosmicCoder96 Actually i have a little doubt regarding that. How can I make that migration file?

**Comment by CosmicCoder96** [comment]
@uds5501 
The correct workflow is as follows:
*  Make changes in the model file which add a column in DB. This won't actually add a column to the database.
* Next, run `python db manage.py db migrate`
* This will generate a migration file
* Finally, run `python manage.py db upgrade`
* This will update the database with the latest migrations.

File: app/api/schema/events.py [hunk 1]
```diff
@@ -118,6 +118,7 @@ def validate_timezone(self, data, original_data):
     refund_policy = fields.String(dump_only=True,
                                   default='All sales are final. No refunds shall be issued in any case.')
     is_stripe_linked = fields.Boolean(dump_only=True, allow_none=True, default=False)
+    is_featured = fields.Bool(default=True)
```
Comment (iamareebjamal) [inline app/api/schema/events.py]: Should be False by default. Do we want every event anyone creates to be featured?

File: migrations/versions/498b78507b64_.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '498b78507b64'
+down_revision = 'e3caa0f2a16c'
+
+
+def upgrade():
+    op.add_column(u'events',sa.Column('is_featured', sa.Boolean(), nullable=True))
```
Comment (iamareebjamal) [inline migrations/versions/498b78507b64_.py]: Shouldn't be nullable

File: app/api/schema/events.py (hunk 1 above)
Comment (uds5501) [reply app/api/schema/events.py]: No, we don't I will fix that

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96  Did as requested

File: migrations/versions/498b78507b64_.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '498b78507b64'
+down_revision = 'e3caa0f2a16c'
+
+
+def upgrade():
+    op.add_column(u'events',sa.Column('is_featured', sa.Boolean(), nullable=False))
```
Comment (iamareebjamal) [inline migrations/versions/498b78507b64_.py]: Are you sure this migration is working? There is no default value

Comment (uds5501) [reply migrations/versions/498b78507b64_.py]: @iamareebjamal It is working in my system, but if you recommend, I can open another PR making the migrations again

Comment (uds5501) [reply migrations/versions/498b78507b64_.py]: @iamareebjamal Actually, I think I will try and migrate again, I shall modify this file accordingly

Comment (uds5501) [reply migrations/versions/498b78507b64_.py]: @iamareebjamal It works, I rerun the migrations and it does return it this way only

**Comment by CosmicCoder96** [comment]
@uds5501 This PR is breaking all the tests on the frontend for Travis, test a migration file before sending in.
The column is defined as non-nullable but, it all the previous events contain null, migrations fails because of that.

File: migrations/versions/498b78507b64_.py (hunk 3 above)
Comment (CosmicCoder96) [reply migrations/versions/498b78507b64_.py]: @iamareebjamal already mentioned this. It's not working, here is the error log on heroku:
```
    exc_info
  File "/app/.heroku/python/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/app/.heroku/python/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 186, in reraise
    raise value.with_traceback(tb)
  File "/app/.heroku/python/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
    context)
  File "/app/.heroku/python/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) column "is_featured" contains null values
 [SQL: 'ALTER TABLE events ADD COLUMN is_featured BOOLEAN NOT NULL']
```

**Comment by uds5501** [comment]
@CosmicCoder96 I will make sure that I submit a tested migration in next PR. I changed the `nullable=False` upon review

**Pull request merged**